src/diffusers/pipelines/dit/pipeline_dit.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2021 OpenAI
+# MIT License
+
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+
+from ...models import AutoencoderKL, Transformer2DModel
+from ...schedulers import DDIMScheduler
+from ...utils import randn_tensor
+from ..pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+
+
+class DiTPipeline(DiffusionPipeline):
+    r"""
+    This pipeline inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
+    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)
+
+    Parameters:
+        dit ([`Transformer2DModel`]):
+            Class conditioned Transformer in Diffusion model to denoise the encoded image latents.
+        vae ([`AutoencoderKL`]):
+            Variational Auto-Encoder (VAE) Model to encode and decode images to and from latent representations.
+        scheduler ([`DDIMScheduler`]):
+            A scheduler to be used in combination with `dit` to denoise the encoded image latents.
+    """
+
+    def __init__(self, dit: Transformer2DModel, vae: AutoencoderKL, scheduler: DDIMScheduler):
```

Comment:
```suggestion
    def __init__(self, transformer: Transformer2DModel, vae: AutoencoderKL, scheduler: DDIMScheduler):
```

IMO `transformer` is a better name here, since DiT is the whole pipeline. Also it's easy to compare to `unet` then for stable diffusion.

Reply:
+1